### DISCUSSION TITLE
Q1, Q2, Q3 in Qualification data

### DISCUSSION THREAD

**Discussion by foxyskippy** [theOehrly/Fast-F1#230] [Q&A]
Hi,

Is there a way to see if a lap has been driven in Q1, Q2 and Q3?

#Script
import fastf1 as ff1
import fastf1.plotting
from fastf1.core import Laps
import numpy as np
import matplotlib as mpl
import pandas as pd

from matplotlib import pyplot as plt
from matplotlib.collections import LineCollection

grandprix = 13
year = 2022

#Qualification
session = ff1.get_session(year, grandprix, 'Q')
weekend = session.event
laps = session.load_laps()

df = pd.DataFrame(laps)

print (df)

DriverNumber
LapTime
LapNumber
Stint
PitOutTime
PitInTime
Sector1Time
Sector2Time
Sector3Time
Sector1SessionTime
Sector2SessionTime
Sector3SessionTime
SpeedI1
SpeedI2
SpeedFL
SpeedST
IsPersonalBest
Compound
TyreLife
FreshTyre
LapStartTime
Team
Driver
TrackStatus
IsAccurate

**Comment by theOehrly** [comment]
There is no "nice" way to do it currently. But `Session.session_status` (https://theoehrly.github.io/Fast-F1/core.html#fastf1.core.Session.session_status) provides all the information about when the individual qualifying sessions start and end. So you can split the Laps data based on these timestamps. 

Better support for this has been discussed but not yet been added.

**Comment by walteij** [comment]
Well, I've been tinkering a bit to get telemetry data and drop it into a MariaDB, and then get PowerBi to work with the data, and I found something that might be interesting to split the Q1/Q2/Q3 times.
[image of the quali data](https://cdn.discordapp.com/attachments/915349296331776030/1099050446452170853/image.png)
So there is a clear distinction between the sessions from the F1 livetiming. Drivers that are out will show -9.223.372.036.854.775.808 in the session.

**Reply by theOehrly** [reply]
I'm not sure what those values actually mean. The representation in Python is different. Anyhow, differentiating between Q1, Q2, Q3 has been implemented based on session status data and will be in the next release.
